File: 4. Median of Two Sorted Arrays/Simpleton Approach/median_of_two_sorted_arrays.swift
func get_median(list: Array<Int>) -> Float{
	let len_list: Int = list.count;
	let even_length: Bool = (len_list % 2 == 0);
	var median: Float = 0.0;
	
	if (even_length){
		median = (Float(list[len_list / 2]) + Float(list[(len_list / 2) - 1])) / 2;
		//median = ((Float)list[len_list / 2] + (Float)list[(len_list / 2) - 1]) / 2;
	} else {
		median = (Float(list[len_list / 2]));
		//median = (Float)list[len_list / 2];
	}
	
	return (median);
}

func combine_list (list1: Array<Int>, list2: Array<Int>) -> Array<Int>{
	var combined_list: Array<Int> = [];
	
	for i in list1{
		combined_list.append(i);
	}
	
	for i in list2{
		combined_list.append(i);
	}
	
	combined_list.sort();
	
	return (combined_list);
}

let list1: Array<Int> = [1,3,5];
let list2: Array<Int> = [2,4];

let list_combined: Array<Int> = combine_list (list1: list1, list2: list2);

let median: Float = get_median(list: list_combined);

print("This is the work of Swift");
print (median);

print ("\n\n\n");

File: 4. Median of Two Sorted Arrays/Simpleton Approach/median_of_two_sorted_arrays.rs
fn get_median (list: Vec<isize>) -> f64 {
	let len_list: usize = list.len();
	let even_length: bool = (len_list % 2 == 0);
	let mut median: f64 = 0.0;
	
	if even_length{
		median = (list[len_list / 2] as f64 + list[(len_list / 2) - 1] as f64) / 2.0;
	} else {
		median = list[len_list / 2] as f64;
	}
	
	return median;
}

fn combine_list (list1: Vec<isize>, list2: Vec<isize>) -> Vec<isize>{
	let len_list1: usize = list1.len();
	let len_list2: usize = list2.len();
	let mut combined_list: Vec<isize> = vec![];
	
	for i in 0..len_list1{
		combined_list.push(list1[i]);
	}
	
	for i in 0..len_list2{
		combined_list.push(list2[i]);
	}
	
	combined_list.sort();
	
	return combined_list;
}

fn main (){
	let list1: Vec<isize> = vec![1,3];
	let list2: Vec<isize> = vec![2,4];
	let mut list_combined: Vec<isize> = combine_list(list1.clone(), list2.clone());
	
	let median: f64 = get_median(list_combined.clone());
	
	println!("This is the handy work of Rust\n\n{}\n\n\n", median);
}
